source/core/slang-char-util.h

```diff
@@ -0,0 +1,51 @@
+#ifndef SLANG_CORE_CHAR_UTIL_H
+#define SLANG_CORE_CHAR_UTIL_H
+
+#include "slang-string.h"
+
+namespace Slang {
+
+struct CharUtil
+{
```

Comment:
We've talked about it many many times, but I will continue to raise my objection that I think the stuff here would be easier to discover and use if it was written as global functions rather than namespaced into a type.